artifactory/utils/container/buildinfo.go

```diff
@@ -0,0 +1,584 @@
+		writer.Write(layer)
+	}
+	filePath = writer.GetFilePath()
+	return
+}
+
+// Create a docker build info.
+func (builder *buildInfoBuilder) createBuildInfo(module string) (*buildinfo.BuildInfo, error) {
+	imageProperties := map[string]string{}
+	imageProperties["docker.image.id"] = builder.imageId
+	imageProperties["docker.image.tag"] = builder.image.Tag()
+	if module == "" {
+		module = builder.image.Name()
+	}
+	buildInfo := &buildinfo.BuildInfo{Modules: []buildinfo.Module{{
+		Id:           module,
+		Type:         buildinfo.Docker,
+		Properties:   imageProperties,
+		Artifacts:    builder.artifacts,
+		Dependencies: builder.dependencies,
+	}}}
+	return buildInfo, nil
+}
+
+// Return - manifest artifacts as buildinfo.Artifact object.
+func getManifestArtifact(searchResults map[string]*utils.ResultItem) (artifact buildinfo.Artifact) {
+	item := searchResults["manifest.json"]
+	return buildinfo.Artifact{Name: "manifest.json", Type: "json", Checksum: &buildinfo.Checksum{Sha1: item.Actual_Sha1, Md5: item.Actual_Md5}, Path: path.Join(item.Repo, item.Path, item.Name)}
+}
+
+// Return - manifest dependency as buildinfo.Dependency object.
+func getManifestDependency(searchResults map[string]*utils.ResultItem) (dependency buildinfo.Dependency) {
+	item := searchResults["manifest.json"]
+	return buildinfo.Dependency{Id: "manifest.json", Type: "json", Checksum: &buildinfo.Checksum{Sha1: item.Actual_Sha1, Md5: item.Actual_Md5}}
+}
+
+// Download and read the config layer from Artifactory.
+// Returned values:
+// configurationLayer - pointer to the configuration layer struct, retrieved from Artifactory.
+// artifact - configuration layer as buildinfo.Artifact struct.
+// dependency - configuration layer as buildinfo.Dependency struct.
+func getConfigLayer(imageId string, searchResults map[string]*utils.ResultItem, serviceManager artifactory.ArtifactoryServicesManager) (configurationLayer *configLayer, artifact buildinfo.Artifact, dependency buildinfo.Dependency, err error) {
+	item := searchResults[digestToLayer(imageId)]
+	configurationLayer = new(configLayer)
+	if err := artutils.RemoteUnmarshal(serviceManager, item.GetItemRelativePath(), &configurationLayer); err != nil {
+		return nil, buildinfo.Artifact{}, buildinfo.Dependency{}, err
+
+	}
+	artifact = buildinfo.Artifact{Name: digestToLayer(imageId), Checksum: &buildinfo.Checksum{Sha1: item.Actual_Sha1, Md5: item.Actual_Md5}, Path: path.Join(item.Repo, item.Path, item.Name)}
+	dependency = buildinfo.Dependency{Id: digestToLayer(imageId), Checksum: &buildinfo.Checksum{Sha1: item.Actual_Sha1, Md5: item.Actual_Md5}}
+	return
+}
+
+func searchManifestAndLayersDetails(builder *buildInfoBuilder, imagePathPattern string) (resultMap map[string]*utils.ResultItem, manifestContent *manifest, err error) {
+	resultMap, err = searchHandler(imagePathPattern, builder)
+	if err != nil || len(resultMap) == 0 {
+		log.Debug("Couldn't find manifest.json. Image path pattern: " + imagePathPattern + ".")
```

Comment:
log.Debug supports multiple arguments. No need for strings concatenation.
Let's changes thsi in other places as well.